sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -163,6 +163,51 @@ final class CastExprMeta[INPUT <: UnaryExpression with TimeZoneAwareExpression w
   override protected val needTimezoneTagging: Boolean = false
 }
 
+trait CastOptions extends Serializable {
+  // The brackets that are used in casting structs and maps to strings
+  def leftBracket: String
+
+  def rightBracket: String
+
+  // The string value to use to represent null elements in array/struct/map.
+  def nullString: String
+
+  def useDecimalPlainString: Boolean
+
+  def useHexFormatForBinary: Boolean
+
+  def stringToDateAnsiModeEnabled: Boolean
```

Comment:
nit: We have three different patterns for boolean parameters here. 

   * useSomething
   * somethingEnabled
   * something

I know we were not consistent before, but it would be good to become consistent so can we pick one and make them all match.

Also could we get javadocs explaining what each means. This is not a sealed trait so in theory anyone could add their own implementation, and it would be good to follow that.